lxqt-session/src/lxqtmodman.cpp

```diff
@@ -345,6 +345,12 @@ void LXQtModuleManager::logout()
         if (p->state() != QProcess::NotRunning && !p->waitForFinished(2000))
         {
             qCWarning(SESSION) << QString("Module '%1' won't terminate ... killing.").arg(i.key());
+
+            // disconnect everything connected to the module signals.
+            // Needed to prevent running LXQtModuleManager::restartModules()
+            // on an already destroyed process and the subsequent Q_ASSERT(proc)
+            // failure/crash
+            p->disconnect();
```

Comment:
OK... I didn't check it correctly, forget my previous comment.

But looking on it for the second time...I don't get it. How can the mentioned Q_ASSERT be violated? Is it possible in Qt to run a slot which was connected to signal of deleted object?